src/jetstream_kv.rs

```diff
@@ -0,0 +1,909 @@
+    /// # let context = nats::jetstream::new(client);
+    /// #
+    /// # let bucket = context.create_key_value(&KeyValueConfig {
+    /// #  bucket: "entry".to_string(),
+    /// #  ..Default::default()
+    /// # })?;
+    /// #
+    /// bucket.put("foo", b"bar")?;
+    /// let maybe_entry = bucket.entry("foo")?;
+    /// if let Some(entry) = maybe_entry {
+    ///   println!("Found entry {:?}", entry);
+    /// }
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn entry(&self, key: &str) -> io::Result<Option<KeyValueEntry>> {
+        if !is_valid_key(key) {
+            return Err(io::Error::new(io::ErrorKind::Other, "invalid key"));
+        }
+
+        let mut subject = String::new();
+        subject.push_str(&self.prefix);
+        subject.push_str(key);
+
+        match self.context.get_last_message(&self.stream_name, &subject) {
+            Ok(message) => {
+                let operation = kv_operation_from_stream_message(&message);
+                let entry = KeyValueEntry {
+                    bucket: self.name.clone(),
+                    key: key.to_string(),
+                    value: message.data,
+                    revision: message.sequence,
+                    created: message.time,
+                    operation,
+                    delta: 0,
+                };
+
+                Ok(Some(entry))
+            }
+            Err(err) => {
+                if let Some(inner_err) = err.get_ref() {
+                    if let Some(error) = inner_err.downcast_ref::<Error>() {
+                        if error.error_code() == ErrorCode::NoMessageFound {
+                            return Ok(None);
+                        }
+                    }
+                }
+
+                Err(err)
+            }
+        }
+    }
+
+    /// Returns the latest value for the key, if any.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use nats::jetstream::KeyValueConfig;
+    /// # fn main() -> std::io::Result<()> {
+    /// # let client = nats::connect("demo.nats.io")?;
+    /// # let context = nats::jetstream::new(client);
+    /// #
+    /// # let bucket = context.create_key_value(&KeyValueConfig {
+    /// #  bucket: "get".to_string(),
+    /// #  ..Default::default()
+    /// # })?;
+    /// #
+    /// bucket.put("foo", b"bar")?;
+    /// let maybe_value = bucket.get("foo")?;
+    /// if let Some(value) = maybe_value {
+    ///   println!("Found value {:?}", value);
+    /// }
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn get(&self, key: &str) -> io::Result<Option<Vec<u8>>> {
+        match self.entry(key) {
+            Ok(Some(entry)) => {
+                let value = if entry.operation == KeyValueOperation::Put {
+                    Some(entry.value)
+                } else {
+                    None
+                };
+
+                Ok(value)
+            }
+            Ok(None) => Ok(None),
+            Err(err) => Err(err),
+        }
```

Comment:
Little hard to read.
Something like that is IMHO clearer to read:

```suggestion
        match self.entry(key) {
            Ok(Some(entry)) => {
                match entry.operation {
                    KeyValueOperation::Put => Ok(Some(entry.value)),
                    _ => Ok(None),
                }
            }
            Ok(None) => Ok(None),
            Err(err) => Err(err),
        }
```

Reply:
Agreed and done.